src/usr/share/netcheck/networkmanagerhelper.py

```diff
@@ -0,0 +1,291 @@
+python-networkmanager class.
+"""
+
+__all__ = ['NetworkManagerHelper']
+__author__ = 'Andrew Klapp and Joel Allen Luellwitz'
+__version__ = '0.8'
+
+import logging
+import time
+import NetworkManager
+
+NETWORKMANAGER_ACTIVATION_CHECK_DELAY = 0.1
+
+NM_CONNECTION_ACTIVATING = "NM_Connection_Activating"
+NM_CONNECTION_ACTIVATED = "NM_Connection_Activated"
+NM_CONNECTION_DISCONNECTED = "NM_Connection_Disconnected"
+
+class DeviceNotFoundException(Exception):
+    """Raised when an interface name passed to NetworkManagerHelper is not found."""
+
+class UnknownConnectionIDException(Exception):
+    """Raised when a connection ID is requested, but it is not known."""
+
+class NetworkManagerHelper:
+    """NetworkManagerHelper abstracts away some of the messy details of the NetworkManager
+    D-Bus API.
+    """
+
+    def __init__(self, config):
+        """Initializes the module by choosing devices and assembling a table of connection
+        IDs.
+
+        config: The configuration dictionary built in netcheck-daemon.
+        """
+
+        self.logger = logging.getLogger(__name__)
+        self.network_activation_timeout = config['network_activation_timeout']
+        self.wired_network_name = config['wired_network_name']
+        self.wifi_network_names = config['wifi_network_names']
+
+        self.connection_id_to_connection_dict = self._build_connection_id_table()
+
+        self.wired_device, self.wireless_device = self._create_device_objects(
+            config['wired_interface_name'],
+            config['wifi_interface_name'])
+
+    def activate_connection(self, connection_id):
+        """Tells NetworkManager to activate a connection with the supplied connection_id.
+        Returns True if there are no errors, False otherwise.
+
+        connection_id: The displayed name of the connection in NetworkManager.
+        """
+        success = False
+
+        if not self.connection_is_activated(connection_id):
+            connection = self.connection_id_to_connection_dict[connection_id]
+            network_device = self._get_device_for_connection(connection)
+
+            networkmanager_output = NetworkManager.NetworkManager.ActivateConnection(
+                connection, network_device, '/')
+
+            try:
+                self._run_proxy_call(networkmanager_output)
+                success = self._wait_for_connection(connection)
+
+            except Exception as exception:
+                self.logger.error('D-Bus call failed: %s', exception)
+
+        else:
+            self.logger.debug('Connection %s is already activated.', connection_id)
+            success = True
+
+        return success
+
+    def get_connection_ip(self, connection_id):
+        """Attempts to retrieve the IP address associated with the given connection's gateway.
+        If the IP address is unable to be retrieved, None is returned.
+
+        connection_id: The displayed name of the connection in NetworkManager.
+        """
```

Comment:
Return type should be explicitly stated. Some of the content from the general comment can be moved here.